### PR TITLE
Task log

### PR DESCRIPTION
**Pull request by bretambrose**
Adds a type string tag to the task struct and logs task scheduling and execution

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/task_scheduler.c [hunk 1]
```diff
@@ -103,6 +138,13 @@ void aws_task_scheduler_schedule_future(
     AWS_ASSERT(task);
     AWS_ASSERT(task->fn);
 
+    AWS_LOGF_DEBUG(
```
Comment (graebm) [inline source/task_scheduler.c]: TRACE

File: source/task_scheduler.c [hunk 2]
```diff
@@ -15,8 +15,37 @@
 
 #include <aws/common/task_scheduler.h>
 
+#include <aws/common/logging.h>
+
+#include <inttypes.h>
+
 static const size_t DEFAULT_QUEUE_SIZE = 7;
 
+const char *aws_task_status_to_c_string(enum aws_task_status status) {
```
Comment (graebm) [inline source/task_scheduler.c]: we already use the term "c_str" in byte_buf.h and string.h
aws_task_status_to_c_str()
or
aws_task_status_c_str()

File: source/task_scheduler.c [hunk 3]
```diff
@@ -15,8 +15,37 @@
...
+            return "<Canceled>";
+
+        default:
+            return "<Unknown>";
+    }
+}
+
+void aws_task_run(struct aws_task *task, enum aws_task_status status) {
+    AWS_ASSERT(task->fn);
+    AWS_LOGF_DEBUG(
```
Comment (graebm) [inline source/task_scheduler.c]: TRACE

File: source/task_scheduler.c [hunk 4]
```diff
@@ -87,6 +116,12 @@ void aws_task_scheduler_schedule_now(struct aws_task_scheduler *scheduler, struc
     AWS_ASSERT(task);
     AWS_ASSERT(task->fn);
 
+    AWS_LOGF_DEBUG(
```
Comment (graebm) [inline source/task_scheduler.c]: TRACE

File: source/task_scheduler.c (hunk 1 above)
Comment (bretambrose) [reply source/task_scheduler.c]: My current rule of thumb is to use TRACE only for debugging information that's too costly (spammy) to log under normal circumstances.  This doesn't feel like it meets that, so I'd prefer to keep it DEBUG for now.  Same for the others.

**Pull request merged**